Userland/Applications/PixelPaint/MainWidget.cpp

```diff
@@ -227,6 +227,17 @@ void MainWidget::initialize_menubar(GUI::Window& window)
         if (!bitmap)
             return;
 
+        auto& image = editor->image();
+        const auto& bitmap_size = bitmap->size();
+        const auto& image_size = image.size();
+        if (bitmap_size.width() > image_size.width() || bitmap_size.height() > image_size.height()) {
+            auto expand_canvas = GUI::MessageBox::show(&window, "Do you want to expand the editor image?", "Expand image?", GUI::MessageBox::Type::Question, GUI::MessageBox::InputType::YesNoCancel);
```

Comment:
1. Popups are usually not ideal. I totally understand why it's useful here, but I mention it in case someone comes up with a way to deal with this without a popup. (Maybe some kind of "suggestion bar" below the toolbar? That would be out of scope though …)
2. Please improve the text. Image you're a user, and just want to paste an image.
    > "Do you want to expand the editor image?"
    > Err, what? No, I want to paste the image!
    > *clicks cancel*
    > Nothing happened? Where is my image?!
    > *presses ctrl-v again*
    > "Do you want to expand the editor image?"
    > No! I want to paste the image, now go away!
    > *clicks no*
    > Oof, why is the image now cropped– Oh! That's why.

    I suggest changing the text to something like "The pasted image is larger than the current editor image. How would you like to proceed?" and making the clickable options something like "Expand editor image", "Paste oversize image" and "Cancel". (Or something along those lines.)

Reply:
As for 2), do you think it would be a good idea to expand `MessageBox` with functionality that allows for adding custom buttons and then use that `MessageBox` or should I just implement a new `Dialog`? I could implement the former if that functionality is desired.

Reply:
Anything that makes sure that pressing "Ctrl-V" doesn't raise a seemingly-unrelated popup. :)